### ISSUE TITLE
[🐞] Context+hidden slot breaks SSR

### ISSUE THREAD

**Issue by wmertens** [BuilderIO/qwik#5270]
### Which component is affected?

Qwik Runtime

### Describe the bug

When SSR renders a child into a parent that doesn't render a Slot, it will not see the parent's context. On the client it's not a problem.

### Reproduction

https://qwik.builder.io/playground/#v=1.2.12&f=7VZLTsMwFNxzCrPpR0UhmwqBmm64AKIL1kkUClLVSI2RqCzfnRl%2FYxeQuqzUXewXv9%2BMPW9EmseHCBTskQB3WAUF5oLEdpsZLdNdrjKa3vxHQdurFo3KyQ2pWs%2Bm7RRJp431gV9qgMcWj3mrcEerkvB1%2B%2Bq93g2dDgC42QMnQmngR7dHBG%2Bku2iEL2P75S62IPL8BFTOSwEaMzPFUUFZJsyGAm4hEJW4dZ9wZT8Wi7ley3673XVh5gruDhFZ5X1MJmJFZGDSDmeXAqFm52xZLnuraGnDsu64BvhnqDX1%2BzeAvp%2Fl95NQLmFt49lY4pz3xDUsw5Jo%2BCIhwgPHR0mlyP5rjhxMJM0RaEgy9LA91MOHGWY2m1ffvKT%2B0EbKD2YUiDmKEgPGJVibTizLQrxRso%2F9F%2BOMY2DSMGHwG1hmeM1YUGbmVTc9ZPtTFj5wljfJqZalTgr9M7%2FVfXrcX%2BXTXYf29TW%2BvNf4Bw

### Steps to reproduce

- remove the second `open` in the playground. The context is now the one from the top parent
- remove the first `open` in the playground. SSR will crash.

### System Info

```shell
qwik v1.2.12
```

### Additional Information

working on a fix

playground code:
```tsx
import {
  component$,
  useSignal,
  Slot,
  useContextProvider,
  useContext,
  createContextId
} from '@builder.io/qwik';

const c = createContextId<any>('c');

export const ProviderParent = component$(({val=0, open=false}) => {
  const s = useSignal(open);
  const t = useSignal(val);
  useContextProvider(c, t)
  return (
    <div>
      <button onClick$={() => (s.value = !s.value, t.value++)}>toggle</button>
      <br />
      {s.value && <Slot />}
    </div>
  );
});

const ContextChild = component$(() => {
  const t = useContext(c);
  return <div>Ctx: {t.value}</div>;
});
 
export default component$(() => {
  return (
    <ProviderParent open>
      Closing this ProviderParent by setting open=false will crash the SSR
      <ContextChild />
      The below Ctx should be 50. When you set open=false it will be 0, from the parent above it.
      <ProviderParent val={50} open>
        <ContextChild />
      </ProviderParent>
    </ProviderParent>
  );
});
```

**Comment by mhevery** [comment]
Any chance you could attempt to create a fix?

**Comment by wmertens** [comment]
Yes, I'm working on it :)

**Comment by mhevery** [comment]
you rock!

**Comment by wmertens** [comment]
This is fixed in #5263
